src/compiler/transformers/module/module.ts

```diff
@@ -3,16 +3,756 @@
+                        createVariableStatement(
+                            /*modifiers*/ undefined,
+                            createVariableDeclarationList([
+                                createVariableDeclaration(
+                                    getSynthesizedClone(namespaceDeclaration.name),
+                                    getGeneratedNameForNode(node),
+                                    /*location*/ node
+                                )
+                            ])
+                        )
+                    );
+                }
+
+                addExportImportAssignments(statements, node);
+                return createNodeArrayNode(statements);
+            }
+
+            return undefined;
+        }
+
+        function visitImportEqualsDeclaration(node: ImportEqualsDeclaration): OneOrMany<Statement> {
+            if (contains(externalImports, node)) {
+                const statements: Statement[] = [];
+                if (moduleKind !== ModuleKind.AMD) {
+                    if (node.flags & NodeFlags.Export) {
+                        addNode(statements,
+                            createStatement(
+                                createExportAssignment(
+                                    node.name,
+                                    createRequireCall(node)
+                                ),
+                                /*location*/ node
+                            )
+                        );
+                    }
+                    else {
+                        addNode(statements,
+                            createVariableStatement(
+                                /*modifiers*/ undefined,
+                                createVariableDeclarationList([
+                                    createVariableDeclaration(
+                                        getSynthesizedClone(node.name),
+                                        createRequireCall(node),
+                                        /*location*/ node
+                                    )
+                                ])
+                            )
+                        );
+                    }
+                }
+                else {
+                    if (node.flags & NodeFlags.Export) {
+                        addNode(statements,
+                            createStatement(
+                                createExportAssignment(node.name, node.name),
+                                /*location*/ node
+                            )
+                        );
+                    }
+                }
+
+                addExportImportAssignments(statements, node);
+                return createNodeArrayNode(statements);
+            }
+
+            return undefined;
+        }
+
+        function visitExportDeclaration(node: ExportDeclaration): OneOrMany<Statement> {
+            if (contains(externalImports, node)) {
+                const generatedName = getGeneratedNameForNode(node);

```

Comment:
* `visitExportDeclaration` must always call `getGeneratedNameForNode`. 
* `visitImportDeclaration` calls it conditionally.
* `visitImportEqualsDeclaration` doesn't call it at all.